packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +36,103 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
+  lineHeight: keyof Theme['typography']['lineHeights']; // prop
+  minHeight: '48px' | '40px' | '32px' | '28px'; // raw value
+  iconSpacing: string; // raw value
+  iconColor: IconProps['color']; // prop
+  textColor: BaseTextProps['color']; // prop
+  spacing: string; // raw value
+  text?: string;
+};
+
+const getProps = ({
+  theme,
+  size,
+  children,
+  buttonTypographyTokens,
+}: {
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+  children?: string;
+  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
+}): BaseButtonStyleProps => {
+  const props: BaseButtonStyleProps = {
+    iconSize: 'medium',
+    fontSize: buttonTypographyTokens.fonts.size.medium,
+    lineHeight: buttonTypographyTokens.lineHeight.medium,
+    minHeight: '40px',
+    iconSpacing: makeSpace(theme.spacing[2]),
+    iconColor: 'action.icon.primary.default',
+    textColor: 'action.text.primary.default',
+    spacing: `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])} ${makeSpace(
+      theme.spacing[0],
+    )} ${makeSpace(theme.spacing[5])}`,
+    text: children?.trim(),
+  };
+
+  switch (size) {
+    case 'large':
+      props.iconSize = 'medium';
+      props.fontSize = buttonTypographyTokens.fonts.size.large;
+      props.lineHeight = buttonTypographyTokens.lineHeight.large;
+      props.minHeight = '48px';
+      props.iconSpacing = makeSpace(theme.spacing[2]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])} ${makeSpace(
+        theme.spacing[0],
+      )} ${makeSpace(theme.spacing[5])}`;
+      break;
+    case 'medium':
+      props.iconSize = 'medium';
+      props.fontSize = buttonTypographyTokens.fonts.size.medium;
+      props.lineHeight = buttonTypographyTokens.lineHeight.medium;
+      props.minHeight = '40px';
+      props.iconSpacing = makeSpace(theme.spacing[2]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])} ${makeSpace(
+        theme.spacing[0],
+      )} ${makeSpace(theme.spacing[5])}`;
+      break;
+    case 'small':
+      props.iconSize = 'xsmall';
+      props.fontSize = buttonTypographyTokens.fonts.size.small;
+      props.lineHeight = buttonTypographyTokens.lineHeight.small;
+      props.minHeight = '32px';
+      props.iconSpacing = makeSpace(theme.spacing[1]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[3])} ${makeSpace(
+        theme.spacing[0],
+      )} ${makeSpace(theme.spacing[3])}`;
+      break;
+    case 'xsmall':
+      props.iconSize = 'xsmall';
+      props.fontSize = buttonTypographyTokens.fonts.size.xsmall;
+      props.lineHeight = buttonTypographyTokens.lineHeight.xsmall;
+      props.minHeight = '28px';
+      props.iconSpacing = makeSpace(theme.spacing[1]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[2])} ${makeSpace(
+        theme.spacing[0],
+      )} ${makeSpace(theme.spacing[2])}`;
```

Comment:
Extract it to a function? 

```ts
  const getSpacing = (one: keyof Theme['spacing'], two: keyof Theme['spacing']): string => {
    return `${makeSpace(theme.spacing[one])} ${makeSpace(theme.spacing[two])} ${makeSpace(
      theme.spacing[one],
    )} ${makeSpace(theme.spacing[two])}`;
  };
```

would make things much cleaner.

Reply:
That would be an overkill imo. Spacing isn't always a 2 input 4 output function. It could be 1, or 2, or 4 output function. Generalizing it would mean that we sometimes use it and sometimes don't. `makeSpace` is already an abstraction. I don't think wrapping it further makes sense. Although I did use the shorthand now and made it only 2 units instead of 4

Reply:
Makes sense but I'm not saying to generalise globally, just locally it will clean things up in the switch statement. 
Though you can make a call here as you find it fitting.